app/components/channel_selector/channel_list.tsx

```diff
@@ -73,11 +82,13 @@ export default function ChannelList({
         return (
             <ChannelListRow
                 channel={item}
+                selectable={itemSelectable}
+                selected={selectedChannels.findIndex(({id}) => id === item.id) !== -1}
                 testID='browse_channels.custom_list.channel_item'
                 onPress={onSelectChannel}
             />
         );
-    }, [onSelectChannel]);
+    }, [onSelectChannel, itemSelectable, selectedChannels]);
```

Comment:
`selectedChannels` will cause this function to change every time you select a channel `onSelectChannel` causing the `useCallback` to not really do anything but cause perf degradation as it also causes the list to re-render as the prop for `renderItem` is changing. you should consider a different approach